backends/guest/common/generate.c

```diff
@@ -449,7 +454,7 @@ int is_x509certificate(const uint8_t *buffer, const size_t buffer_size, uint8_t
  * @param buffer_size , length of buffer
  * @param hash_funct, index of hash function information to use for ESL GUID,
  *                   also helps in prevalation, if inform is '[c]ert' then this doesn't matter
- * @param hash_data, the generated hash data
+ * @param hash_data, the generated hash data, buffer should be allocated before calling
```

Comment:
Not worth going into for this PR, but on the topic of function names matching what they do: I don't like that a function implying boolean with `is_` was apparently allocating memory. Arguably this should be renamed to `try_`, or actually be made a boolean function that does not actually do any type conversion.

I'll add this to the backlog, but we need to make a pass on function names and behaviors.